test/realtime_web/controllers/channels_controller_test.exs

```diff
@@ -86,60 +94,81 @@ defmodule RealtimeWeb.ChannelsControllerTest do
-  #     conn = post(conn, ~p"/v3/api/channels", %{})
-  #     assert json_response(conn, 422) == %{"errors" => %{"name" => ["can't be blank"]}}
-  #   end
-  # end
-
-  # describe "delete" do
-  #   test "deletes a channel", %{conn: conn, tenant: tenant} do
-  #     channel = channel_fixture(tenant)
-  #     conn = delete(conn, ~p"/v3/api/channels/#{channel.id}")
-  #     assert conn.status == 202
-  #   end
-
-  #   test "returns not found if id doesn't exist", %{conn: conn} do
-  #     conn = delete(conn, ~p"/v3/api/channels/0")
-  #     assert conn.status == 404
-  #   end
-  # end
-
-  # describe "update" do
-  #   test "updates a channel", %{conn: conn, tenant: tenant} do
-  #     channel = channel_fixture(tenant)
-  #     name = random_string()
-  #     conn = put(conn, ~p"/v3/api/channels/#{channel.id}", %{name: name})
-  #     res = json_response(conn, 202)
-  #     assert name == res["name"]
-
-  #     name = random_string()
-  #     conn = patch(conn, ~p"/v3/api/channels/#{channel.id}", %{name: name})
-  #     res = json_response(conn, 202)
-  #     assert name == res["name"]
-  #   end
-
-  #   test "422 if params are invalid", %{conn: conn, tenant: tenant} do
-  #     channel = channel_fixture(tenant)
-  #     conn = put(conn, ~p"/v3/api/channels/#{channel.id}", %{name: 1})
-  #     assert json_response(conn, 422) == %{"errors" => %{"name" => ["is invalid"]}}
-  #     conn = patch(conn, ~p"/v3/api/channels/#{channel.id}", %{name: 1})
-  #     assert json_response(conn, 422) == %{"errors" => %{"name" => ["is invalid"]}}
-  #   end
-  # end
+  describe "create" do
+    @tag role: "service_role"
+    test "creates a channel", %{conn: conn} do
+      name = random_string()
+      conn = post(conn, ~p"/v3/api/channels", %{"name" => name})
+      res = json_response(conn, 201)
+      assert name == res["name"]
+    end
+
+    @tag role: "service_role"
+    test "422 if params are invalid", %{conn: conn, db_conn: db_conn} do
+      name = random_string()
+      Channels.create_channel(%{"name" => name}, db_conn)
+      conn = post(conn, ~p"/v3/api/channels", %{"name" => name})
+      assert json_response(conn, 422) == %{"errors" => %{"name" => ["has already been taken"]}}
+    end
+
+    @tag role: "authenticated"
+    test "returns 401 if unauthorized", %{conn: conn} do
+      conn = post(conn, ~p"/v3/api/channels", %{name: random_string()})
+      assert json_response(conn, 401) == %{"message" => "Unauthorized"}
+    end
+  end
+
+  describe "delete" do
+    @tag role: "service_role"
+    test "deletes a channel", %{conn: conn, tenant: tenant} do
+      channel = channel_fixture(tenant)
+      conn = delete(conn, ~p"/v3/api/channels/#{channel.id}")
+      assert conn.status == 202
+    end
+
+    @tag role: "service_role"
+    test "returns unauthorized if id doesn't exist", %{conn: conn} do
+      conn = delete(conn, ~p"/v3/api/channels/0")
+      assert conn.status == 401
+    end
+
+    @tag role: "authenticated"
```

Comment:
Should we have a test where `authenticated` users can create channels if RLS is setup correctly?

Reply:
actually I'm thinking of changing everything to authenticated instead of service role 👀